service/controller/resource/app/desired.go

```diff
@@ -145,17 +151,40 @@ func (r *Resource) newApp(cc controllercontext.Context, cr apiv1alpha2.Cluster,
 	}
 }
 
-func (r *Resource) newAppSpecs() []key.AppSpec {
-	switch r.provider {
-	case "aws":
-		return append(key.CommonAppSpecs(), key.AWSAppSpecs()...)
-	case "azure":
-		return append(key.CommonAppSpecs(), key.AzureAppSpecs()...)
-	case "kvm":
-		return append(key.CommonAppSpecs(), key.KVMAppSpecs()...)
-	default:
-		return key.CommonAppSpecs()
+func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]key.AppSpec, error) {
+	req := searcher.Request{
+		ReleaseVersion: key.ReleaseVersion(&cr),
+	}
+
+	res, err := r.clusterClient.Release.Searcher.Search(ctx, req)
```

Comment:
There is already an operatorversions resource that does this. It fetches the data and adds it to the controller context.

I think this resource should be renamed to `releaseversions` and add both the app and operator versions to the context. So that we make a single API call to cluster-service.

cc @xh3b4sd Since I believe you added the `operatorversions` resource.